store/todos.js

```diff
@@ -7,24 +8,63 @@ export const getters = {
   getTodoList: (state) => {
     return state.taskList;
   },
+  getCompleteRequest: (state) => {
+    return state.completeRequest;
+  },
 };
 
 export const actions = {
   addTask: ({ state, commit }, val) => {
     commit('addTask', val);
   },
   deleteTask: ({ state, commit }, val) => {
-    commit('deleteTask', val);
+    commit('setCompleteRequest', true);
+
+    return new Promise((resolve, reject) => {
+      // return to the location where is was dispatched after being resolved
+      setTimeout(() => {
+        commit('deleteTask', val);
+        commit('setCompleteRequest', false);
+        resolve();
+      }, 1000);
+    });
+  },
+  changeTaskState: ({ state, commit }, val) => {
+    commit('setCompleteRequest', true);
+
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        commit('changeTaskState', val);
+        commit('setCompleteRequest', false);
+        resolve();
+      }, 1000);
+    });
```

Comment:
Why are we returning promise?

Reply:
another functionality depends on this operation